File: Netty/netty08-thrift/src/main/java/com/ayy/server/ThriftServer.java
package com.ayy.server;

import com.ayy.service.PersonServiceImpl;
import com.ayy.thrift.generated.PersonService;
import org.apache.thrift.TProcessorFactory;
import org.apache.thrift.protocol.TCompactProtocol;
import org.apache.thrift.server.THsHaServer;
import org.apache.thrift.server.TServer;
import org.apache.thrift.transport.TFramedTransport;
import org.apache.thrift.transport.TNonblockingServerSocket;

/**
 * @ Description
 * @ Author Zhao JIN
 * @ Date 13/05/2021
 * @ Version 1.0
 */

public class ThriftServer {
    public static void main(String[] args) throws Exception {
        TNonblockingServerSocket serverSocket = new TNonblockingServerSocket(10000);
        THsHaServer.Args arg = new THsHaServer.Args(serverSocket).minWorkerThreads(2).maxWorkerThreads(4);
        PersonService.Processor<PersonServiceImpl> processor = new PersonService.Processor<>(new PersonServiceImpl());
        arg.protocolFactory(new TCompactProtocol.Factory());
        arg.transportFactory(new TFramedTransport.Factory());
        arg.processorFactory(new TProcessorFactory(processor));

        TServer server = new THsHaServer(arg);
        System.out.println("Thrift Server Started");

        server.serve();
    }
}

File: Netty/netty08-thrift/py/com/ayy/thrift/py/server/ThriftServer.py
# _*_ coding:utf-8 _*_
__author__ = "Zhao JIN"

from thrift import Thrift
from com.ayy.thrift.py.generated import PersonService
from com.ayy.thrift.py.handler.PersonServiceHandler import PersonServiceHandler
from thrift.transport import TTransport
from thrift.protocol import TCompactProtocol
from thrift.transport import TSocket
from thrift.server import TServer

try:
    personServiceHandler = PersonServiceHandler()
    processor = PersonService.Processor(personServiceHandler)
    serverSocket = TSocket.TServerSocket(host="127.0.0.1",port=10000)

    transportFactory = TTransport.TFramedTransportFactory()
    protocolFactory = TCompactProtocol.TCompactProtocolFactory()

    server = TServer.TSimpleServer(processor, serverSocket, transportFactory, protocolFactory)

    server.serve()
except Thrift.TException as ex:
    print("%s" % ex.message)